packages/notifications/src/NotificationPortal/NotificationPortal.tsx

```diff
@@ -1,28 +1,37 @@
-import { connect } from 'react-redux';
-import { ActionCreator, Dispatch } from 'redux';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import Portal, { PortalNotificationConfig } from '../Portal';
 import { clearNotifications, removeNotification } from '../redux/actions/notifications';
+import { ErrorBoundary } from '@redhat-cloud-services/frontend-components/ErrorBoundary';
 
 export interface NotificationsState {
-  [key: string]: any;
   notifications?: PortalNotificationConfig[];
 }
 
 export interface NotificationPortalProps {
-  removeNotification: ActionCreator<PortalNotificationConfig>;
-  clearNotifications: ActionCreator<Record<string, unknown>>;
+  clearNotifications?: () => void;
 }
 
-const mapStateToProps = ({ notifications }: NotificationsState, initialProps: NotificationsState) => ({
-  notifications: initialProps.notifications || notifications,
-});
+export const NotificationPortalBase: React.FC<NotificationPortalProps> = ({ clearNotifications: propsClear, ...props }) => {
+  const notifications = useSelector<NotificationsState>(({ notifications }) => notifications);
+  const dispatch = useDispatch();
+  const removeNotif = (id: number | string) => dispatch(removeNotification(id));
+  const onClearAll = () => dispatch(clearNotifications());
+  return (
+    <Portal
+      notifications={notifications as PortalNotificationConfig[]}
+      removeNotification={removeNotif}
+      onClearAll={propsClear || onClearAll}
+      {...props}
+    />
+  );
+};
 
-const mapDispatchToProps = (dispatch: Dispatch, initialProps: NotificationPortalProps) => ({
-  removeNotification: initialProps.removeNotification ? initialProps.removeNotification : (id: number | string) => dispatch(removeNotification(id)),
-  onClearAll: initialProps.clearNotifications ? initialProps.clearNotifications : () => dispatch(clearNotifications()),
-});
-
-export const NotificationPortal = connect<NotificationsState>(mapStateToProps, mapDispatchToProps)(Portal);
+export const NotificationPortal: React.FC<NotificationPortalProps> = (props) => (
+  <ErrorBoundary headerTitle="Notifications portal" silent>
```

Comment:
Can we make it possible to override `silent` by the props from `NotificationPortal`? So that apps that import the portal can also have an ability to modify this behavior...

Reply:
Sure thing, it'll show generic error component above your UI.